server/src/main/java/org/elasticsearch/gateway/LocalAllocateDangledIndices.java

```diff
@@ -123,13 +123,12 @@ public ClusterState execute(ClusterState currentState) {
                         allocationService.getShardRoutingRoleStrategy(),
                         currentState.routingTable()
                     );
-                    final Version minIndexCompatibilityVersion = currentState.getNodes()
-                        .getMaxNodeVersion()
-                        .minimumIndexCompatibilityVersion();
+                    IndexVersion minIndexCompatibilityVersion = currentState.getNodes().getMinSupportedIndexVersion();
+                    IndexVersion maxIndexCompatibilityVersion = currentState.nodes().getMaxDataNodeCompatibleIndexVersion();
```

Comment:
Why the mix of using nodes() here but getNodes() above? We should probably consolidate these methods...(not here, just a thought out loud).